.github/workflows/build.yml

```diff
@@ -0,0 +1,222 @@
+name: CI Build
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+# cancel already running builds of the same branch or pull request
+concurrency:
+  group: ci-${{ github.head_ref }} || concat(${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CFLAGS: -g -O2 -Werror=pointer-arith -Werror=implicit-function-declaration
+  JOBS: 2
+  DEBUG: 1
+  LIBRARY_DEPS: |
+    libatk1.0-dev
+    libcairo2-dev
+    libdconf-dev
+    libgirepository1.0-dev
+    libglib2.0-dev
+    libgtk-3-dev
+    libgtk-layer-shell-dev
+    libice-dev
+    libmate-desktop-dev
+    libmate-menu-dev
+    libmateweather-dev
+    libpango1.0-dev
+    libsm-dev
+    libsoup2.4-dev
+    libwnck-3-dev
+    libx11-dev
+    libxrandr-dev
+    libgirepository1.0-dev
+  BUILD_DEPS: |
+    ccache
+    autoconf-archive
+    autopoint
+    gir1.2-freedesktop
+    git
+    gobject-introspection
+    gtk-doc-tools
+    lsb-release
+    make
+    mate-common
+    meson
+    yelp-tools
+    iso-codes
+    gobject-introspection
+  # TODO
+  SCAN_BUILD_DEPS: |
+    clang
+    clang-tools
+
+jobs:
+  build:
+    name: Build on Ubuntu
+    runs-on: ubuntu-latest
+
+    env:
+      CC: ccache gcc
+      CXX: ccache g++
+      BASE_INSTALL_PATH: ${{github.workspace}}/_install
+
+    steps:
+      - name: Setup environment
+        run: |
+          echo "PATH=${BASE_INSTALL_PATH}/bin:${PATH}" >> "$GITHUB_ENV"
+          echo "PKG_CONFIG_PATH=${BASE_INSTALL_PATH}/lib/pkgconfig:${PKG_CONFIG_PATH}" >> "$GITHUB_ENV"
+
+      - name: Show environment
+        run: env | sort
+        if: ${{ env.DEBUG == '1' }}
+
+      - uses: actions/checkout@v3
+        with:
+          path: mate-panel
+          submodules: true
+
+      # TODO: cache mate-desktop build/install?
+      - name: Checkout mate-desktop
+        uses: actions/checkout@v3
+        with:
+          repository: mate-desktop/mate-desktop
+          ref: v1.27.1
+          path: mate-desktop
+          submodules: true
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install --assume-yes --no-install-recommends \
+            ${BUILD_DEPS} ${LIBRARY_DEPS}
+
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+
+      - name: Configure mate-desktop
+        run: |
+          cd mate-desktop
+          NOCONFIGURE=1 ./autogen.sh
+          { ./configure --prefix="$BASE_INSTALL_PATH" || { cat config.log; exit 1; } ; }
+
+      - name: Build mate-desktop
+        run: make -C mate-desktop -j ${{ env.JOBS }}
+
+      - name: Install mate-desktop
+        run: make -C mate-desktop -j ${{ env.JOBS }} install
+
+      - name: Configure
+        run: |
+          cd mate-panel
+          NOCONFIGURE=1 ./autogen.sh
```

Comment:
Can we use `--enable-compile-warnings=maximum` when building on master branch? Same as we did for travis. https://github.com/mate-desktop/mate-panel/blob/master/.build.yml#L175
For stable branches we used normal compilor warn flags.